server/odc-service/src/main/java/com/oceanbase/odc/service/pldebug/session/AbstractDebugSession.java

```diff
@@ -113,19 +117,50 @@ protected void acquireNewConnection(ConnectionSession connectionSession,
 
     protected DebugDataSource acquireDataSource(ConnectionSession connectionSession, List<String> initSqls) {
         ConnectionConfig config = (ConnectionConfig) ConnectionSessionUtil.getConnectionConfig(connectionSession);
-        DebugDataSource dataSource = new DebugDataSource(config, initSqls);
         String schema = ConnectionSessionUtil.getCurrentSchema(connectionSession);
-        String host;
-        Integer port;
-        if (StringUtils.isBlank(config.getClusterName())) {
-            host = config.getHost();
-            port = config.getPort();
-        } else {
+        String host = config.getHost();
+        Integer port = config.getPort();
+        String url;
+        if (connectionSession.getConnectType() == ConnectType.OB_ORACLE) {
+            if (StringUtils.isBlank(config.getClusterName())
+                    && connectionSession.getConnectType() != ConnectType.CLOUD_OB_ORACLE) {
+                // current connection is a direct observer
+                url = String.format("jdbc:%s://%s:%d/\"%s\"", OB_JDBC_PROTOCOL, host, port, schema);
+                return buildDataSource(config, initSqls, null, url);
+            } else {
+                // obtain one of the proxied observers by odp
+                String directServerIp = getDirectServerIp(connectionSession);
+                String[] ipParts = directServerIp.split(":");
+                host = ipParts[0];
+                port = Integer.parseInt(ipParts[1]);
+                String obProxyVersion = ConnectionSessionUtil.getObProxyVersion(connectionSession);
+                if (ConnectionSessionUtil.isSupportObProxyRoute(obProxyVersion)) {
+                    // use the specified routing function of odp
+                    this.plDebugODPSpecifiedRoute = new PLDebugODPSpecifiedRoute(host, port);
+                    url = String.format("jdbc:%s://%s:%d/\"%s\"", OB_JDBC_PROTOCOL, config.getHost(), config.getPort(),
+                            schema);
+                    return buildDataSource(config, initSqls, this.plDebugODPSpecifiedRoute, url);
+                }
+                // use direct connection observer
+                url = String.format("jdbc:%s://%s:%d/\"%s\"", OB_JDBC_PROTOCOL, host, port, schema);
+                return buildDataSource(config, initSqls, null, url);
+            }
+        } else if (connectionSession.getConnectType() == ConnectType.CLOUD_OB_ORACLE) {
             String directServerIp = getDirectServerIp(connectionSession);
-            host = directServerIp.split(":")[0];
-            port = Integer.parseInt(directServerIp.split(":")[1]);
+            String[] ipParts = directServerIp.split(":");
+            host = ipParts[0];
+            port = Integer.parseInt(ipParts[1]);
+            // use the specified routing function of odp
+            this.plDebugODPSpecifiedRoute = new PLDebugODPSpecifiedRoute(host, port);
```

Comment:
Why no need to judge the odp version here?

Reply:
cloud ob oracle's data source is previously verified in DBPLOperators#create, so it is not needed here.
![image](https://github.com/user-attachments/assets/433c6a68-2523-4765-9358-a1e2d6f10934)

Reply:
It's too tricky and hard to maintain. Put the version check logic together.